test/integration/src/test/java/io/pravega/test/integration/ReaderGroupStreamCutUpdateTest.java

```diff
@@ -0,0 +1,162 @@
+        serviceBuilder.close();
+        zkTestServer.close();
+    }
+
+    @Test
+    public void testStreamcutsUpdateInReaderGroup() throws Exception {
+        final String scope = "testStreamcutsUpdateInReaderGroup";
+        final String stream = "myStream2";
+        final String readerGroupName = "testStreamcutsUpdateInReaderGroupRG";
+        final int checkpointingIntervalMs = 2000;
+        final int readerSleepInterval = 250;
+        final int numEvents = 200;
+
+        // First, create the stream.
+        StreamManager streamManager = StreamManager.create(controllerURI);
+        Assert.assertTrue(streamManager.createScope(scope));
+        StreamConfiguration streamConfiguration = StreamConfiguration.builder()
+                                                                     .scope(scope)
+                                                                     .streamName(stream)
+                                                                     .scalingPolicy(ScalingPolicy.fixed(2))
+                                                                     .build();
+        streamManager.createStream(scope, stream, streamConfiguration);
+
+        // Write some events in the stream.
+        @Cleanup
+        ClientFactory clientFactory = ClientFactory.withScope(scope, controllerURI);
+        writeEvents(clientFactory, stream, numEvents);
+
+        // Read the events and test that positions are getting updated.
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                                                               .stream(Stream.of(scope, stream))
+                                                               .automaticCheckpointIntervalMillis(checkpointingIntervalMs)
+                                                               .build();
+
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, controllerURI);
+        readerGroupManager.createReaderGroup(readerGroupName, readerGroupConfig);
+        ReaderGroup readerGroup = readerGroupManager.getReaderGroup(readerGroupName);
+        @Cleanup
+        EventStreamReader<Double> reader = clientFactory.createReader("myReader", readerGroupName,
+                new JavaSerializer<>(), ReaderConfig.builder().build());
+
+        Map<Stream, StreamCut> currentStreamcuts = readerGroup.getStreamCuts();
+        EventRead eventRead;
+        int lastIteration = 0, iteration = 0;
+        int assertionFrequency = checkpointingIntervalMs / readerSleepInterval;
+        do {
+            eventRead = reader.readNextEvent(5000);
+            if (iteration != lastIteration && iteration % assertionFrequency == 0) {
+                log.info("Comparing streamcuts: {} / {} in iteration {}.", currentStreamcuts, readerGroup.getStreamCuts(), iteration);
+                Assert.assertNotEquals(currentStreamcuts, readerGroup.getStreamCuts());
+                currentStreamcuts = readerGroup.getStreamCuts();
```

Comment:
The test creates only one reader. Hence, if `eventRead.isCheckPoint()` is `true` then the value of `eventRead.getPosition().asImpl().getOwnedSegmentsWithOffsets()` and `readerGroup.getStreamCuts()` should be the same.

Reply:
Unfortunately, `eventRead.getPosition().asImpl().getOwnedSegmentsWithOffsets()` has package-private visibility, so it cannot be accessed from the integration tests package.